datadog/resource_datadog_authn_mapping.go

```diff
@@ -0,0 +1,203 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+func resourceDatadogAuthnMapping() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog AuthN Mappings resource. This can be used to create and manage Datadog SAML Mappings.",
+		CreateContext: resourceDatadogAuthnMappingCreate,
+		ReadContext:   resourceDatadogAuthnMappingRead,
+		UpdateContext: resourceDatadogAuthnMappingUpdate,
+		DeleteContext: resourceDatadogAuthnMappingDelete,
+
```

Comment:
Nice feature to consider adding is support for `terraform import`. Then you can take AuthN mapping defined elsewhere, for example in Datadog's UI, and import them into Terraform like this:

```
terraform import datadog_authn_mapping.dev_ro_mapping "5455f42a-8b9f-11ec-813e-da7ad0900002"
```

As-is, it'll fail with an error `Error: resource datadog_authn_mapping doesn't support import`.

Some other resources but not all implement this feature -- grep for `Importer:`

OK with me if we postpone this feature til later

Reply:
Will look into this. Will include it in this PR if its not a huge task

Reply:
Added support for import